src/main/mondrian/rolap/RolapNativeFilter.java

```diff
@@ -227,6 +228,15 @@ NativeEvaluator createEvaluator(
             evaluator.restore(savepoint);
         }
     }
+
+    private void alertNonNative(
+        RolapEvaluator evaluator, FunDef fun, String reason)
+    {
+        if (!evaluator.getQuery().shouldAlertForNonNative(fun)) {
+            return;
+        }
+        RolapUtil.alertNonNative(getClass().getSimpleName(), reason);
+    }
```

Comment:
Would be better to either hard-code the function name here, or better--just use fun.getName().  Users reviewing the logs will find that more intelligible than the classname.